values/binary.go

```diff
@@ -89,6 +94,16 @@ var binaryFuncLookup = map[BinaryFuncSignature]BinaryFunction{
 		r := rv.Float()
 		return NewFloat(l - r), nil
 	},
+	{Operator: ast.SubtractionOperator, Left: semantic.Time, Right: semantic.Time}: func(lv, rv Value) (Value, error) {
+		l := lv.Time()
+		r := rv.Time()
+		return NewDuration(Duration(l - r)), nil
```

Comment:
We will actually need an new binary operator to specifically handle subtraction of time values. This is because subtracting times produces a duration, but the subtraction operator, along with all of our binary arithmetic operators, is defined to return the same type as that of its operands. See the following https://github.com/influxdata/flux/issues/761#issuecomment-514718381.

My suggestion would be to remove this operation for right now until the type system can support it.